libs/dal/src/repositories/message/message.repository.ts

```diff
@@ -96,12 +96,71 @@ export class MessageRepository extends BaseRepository<MessageDBModel, MessageEnt
       sort: '-createdAt',
     })
       .read('secondaryPreferred')
+      .populate('template', '_id tags')
       .populate('subscriber', '_id firstName lastName avatar subscriberId')
       .populate('actorSubscriber', '_id firstName lastName avatar subscriberId');
 
     return this.mapEntities(messages);
   }
 
+  async paginate(
+    {
+      environmentId,
+      channel,
+      subscriberId,
+      tags,
+      read,
+      archived,
+    }: {
+      environmentId: string;
+      subscriberId: string;
+      channel: ChannelTypeEnum;
+      tags?: string[];
+      read?: boolean;
+      archived?: boolean;
+    },
+    options: { limit: number; after: number | string }
+  ) {
+    const query: MessageQuery & EnforceEnvId = {
+      _environmentId: environmentId,
+      _subscriberId: subscriberId,
+      channel,
+    };
+
+    if (tags && tags?.length > 0) {
+      query.tags = { $in: tags };
+    }
+
+    if (typeof read === 'boolean') {
+      query.read = read;
+    } else {
+      query.read = { $in: [true, false] };
+    }
+
+    if (typeof archived === 'boolean') {
+      if (!archived) {
+        query.$or = [{ archived: { $exists: false } }, { archived: false }];
+      } else {
+        query.archived = true;
+      }
+    } else {
+      query.$or = [{ archived: { $exists: false } }, { archived: { $in: [true, false] } }];
+    }
```

Comment:
Handle the `archive` filtering, the thing is that it's a new field that is by default `undefined` for all the documents, which means the same as `archived: false`.